markdownhighlighter.cpp

```diff
@@ -193,72 +194,72 @@ void MarkdownHighlighter::initHighlightingRules() {
+        rule.pattern = QRegularExpression(QStringLiteral(R"(<(note:\/\/[^\s]+)>)"));
         rule.capturingGroup = 1;
         _highlightingRulesAfter.append(rule);
 
         // highlight links with <> that have a .in it
         //    rule.pattern = QRegularExpression("<(.+?:\\/\\/.+?)>");
-        rule.pattern = QRegularExpression("<([^\\s`][^`]*?\\.[^`]*?[^\\s`])>");
+        rule.pattern = QRegularExpression(QStringLiteral("<([^\\s`][^`]*?\\.[^`]*?[^\\s`])>"));
         rule.capturingGroup = 1;
         _highlightingRulesAfter.append(rule);
 
         // highlight urls with title
         //    rule.pattern = QRegularExpression("\\[(.+?)\\]\\(.+?://.+?\\)");
         //    rule.pattern = QRegularExpression("\\[(.+?)\\]\\(.+\\)\\B");
-        rule.pattern = QRegularExpression(R"(\[([^\[\]]+)\]\((\S+|.+?)\)\B)");
+        rule.pattern = QRegularExpression(QStringLiteral(R"(\[([^\[\]]+)\]\((\S+|.+?)\)\B)"));
         _highlightingRulesAfter.append(rule);
 
         // highlight urls with empty title
         //    rule.pattern = QRegularExpression("\\[\\]\\((.+?://.+?)\\)");
-        rule.pattern = QRegularExpression(R"(\[\]\((.+?)\))");
+        rule.pattern = QRegularExpression(QStringLiteral(R"(\[\]\((.+?)\))"));
         _highlightingRulesAfter.append(rule);
 
         // highlight email links
-        rule.pattern = QRegularExpression("<(.+?@.+?)>");
+        rule.pattern = QRegularExpression(QStringLiteral("<(.+?@.+?)>"));
         _highlightingRulesAfter.append(rule);
 
         // highlight reference links
-        rule.pattern = QRegularExpression(R"(\[(.+?)\]\[.+?\])");
+        rule.pattern = QRegularExpression(QStringLiteral(R"(\[(.+?)\]\[.+?\])"));
         _highlightingRulesAfter.append(rule);
     }
 
     // Images
     {
         // highlight images with text
         HighlightingRule rule(HighlighterState::Image);
-        rule.pattern = QRegularExpression(R"(!\[(.+?)\]\(.+?\))");
+        rule.pattern = QRegularExpression(QStringLiteral(R"(!\[(.+?)\]\(.+?\))"));
         rule.capturingGroup = 1;
         _highlightingRulesAfter.append(rule);
 
         // highlight images without text
-        rule.pattern = QRegularExpression(R"(!\[\]\((.+?)\))");
+        rule.pattern = QRegularExpression(QStringLiteral(R"(!\[\]\((.+?)\))"));
         _highlightingRulesAfter.append(rule);
     }
 
     // highlight images links
     {
 //        HighlightingRule rule;
         HighlightingRule rule(HighlighterState::Link);
-        rule.pattern = QRegularExpression(R"(\[!\[(.+?)\]\(.+?\)\]\(.+?\))");
+        rule.pattern = QRegularExpression(QStringLiteral(R"(\[!\[(.+?)\]\(.+?\)\]\(.+?\))"));
         rule.capturingGroup = 1;
         _highlightingRulesAfter.append(rule);
 
         // highlight images links without text
-        rule.pattern = QRegularExpression(R"(\[!\[\]\(.+?\)\]\((.+?)\))");
+        rule.pattern = QRegularExpression(QStringLiteral(R"(\[!\[\]\(.+?\)\]\((.+?)\))"));
```

Comment:
Does it really work to implant a `R"()"` regular expression into QStringLiteral()?

Reply:
Yeah it works. We already have some of these in QON and nothing is amiss.